MediaBrowser.Controller/Sorting/AlphanumComparator.cs

```diff
@@ -1,94 +1,132 @@
                 {
-                    return 1;
+                    pos2++;
                 }
-                char thisCh = s1[thisMarker];
-                char thatCh = s2[thatMarker];
 
-                var thisChunk = new StringBuilder();
-                var thatChunk = new StringBuilder();
-                bool thisNumeric = char.IsDigit(thisCh), thatNumeric = char.IsDigit(thatCh);
+                var span1 = s1.AsSpan(start1, pos1 - start1);
+                var span2 = s2.AsSpan(start2, pos2 - start2);
 
-                while (thisMarker < s1.Length && char.IsDigit(thisCh) == thisNumeric)
+                if (isNum1 && isNum2)
                 {
-                    thisChunk.Append(thisCh);
-                    thisMarker++;
-
-                    if (thisMarker < s1.Length)
+                    // Trim leading zeros so we can compare the length
+                    // of the strings to find the largest number
+                    span1 = span1.TrimStart('0');
+                    span2 = span2.TrimStart('0');
+                    var span1Len = span1.Length;
+                    var span2Len = span2.Length;
+                    if (span1Len < span2Len)
                     {
-                        thisCh = s1[thisMarker];
+                        return -1;
                     }
-                }
-
-                while (thatMarker < s2.Length && char.IsDigit(thatCh) == thatNumeric)
-                {
-                    thatChunk.Append(thatCh);
-                    thatMarker++;
-
-                    if (thatMarker < s2.Length)
+                    else if (span1Len > span2Len)
                     {
-                        thatCh = s2[thatMarker];
+                        return 1;
                     }
-                }
+                    else if (span1Len >= 20) // Number is probably too big for a ulong
+                    {
+                        // Trim all the first digits that are the same
+                        int i = 0;
+                        while (i < span1Len && span1[i] == span2[i])
+                        {
+                            i++;
+                        }
 
+                        // If there are no more digits it's the same number
+                        if (i == span1Len)
+                        {
+                            continue;
+                        }
 
-                // If both chunks contain numeric characters, sort them numerically
-                if (thisNumeric && thatNumeric)
-                {
-                    if (!int.TryParse(thisChunk.ToString(), out thisNumericChunk)
-                        || !int.TryParse(thatChunk.ToString(), out thatNumericChunk))
+                        // Only need to compare the most significant digit
+                        span1 = span1.Slice(i, 1);
+                        span2 = span2.Slice(i, 1);
+                    }
+
+                    if (!ulong.TryParse(span1, out var num1)
```

Comment:
why parse single digit into `ulong`?

Reply:
It's not always a single digit, only if the number has >19 digits